src/me/wheezygold/skLib/skript/ExprFormatDateOfPlayer.java

```diff
@@ -1,74 +1,53 @@
 package me.wheezygold.skLib.skript;
 
+import ch.njol.skript.Skript;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import me.wheezygold.skLib.common.JsonReader;
+import org.bukkit.entity.Player;
+import org.json.JSONException;
+import org.json.JSONObject;
+
 import java.io.IOException;
 import java.net.InetAddress;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.TimeZone;
 
-import org.bukkit.entity.Player;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-import org.json.JSONException;
-import org.json.JSONObject;
-
-import ch.njol.skript.Skript;
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.util.Kleenean;
-import me.wheezygold.skLib.common.JsonReader;
-
-public class ExprFormatDateOfPlayer extends SimpleExpression<String> {
-	
+public class ExprFormatDateOfPlayer extends SimplePropertyExpression<Player, String>{
 	static {
-		Skript.registerExpression(
-				ExprFormatDateOfPlayer.class, String.class, ExpressionType.SIMPLE, 
-				"[the] formatted date (of|for) [player] %player%");
+		PropertyExpression.register(ExprFormatDateOfPlayer.class, String.class, "formatted date", "player");
 	}
-	
-	private Expression<Player> player;
-
-	@Override
-	public Class<? extends String> getReturnType() {
-		return String.class;
-	}
-
 	@Override
-	public boolean isSingle() {
-		return true;
+	protected String getPropertyName() {
+		return "formatted date";
 	}
 
-	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(Expression<?>[] arg0, int arg1, Kleenean arg2, ParseResult arg3) {
-		player = (Expression<Player>) arg0[0];
-		return true;
+	public String convert(Object o) {
+		if(o instanceof Player){
+			InetAddress rawip = ((Player) o).getAddress().getAddress();
```

Comment:
There should be no problem on changing that while you're doing this pull request though, if you don't mind.